### ISSUE TITLE
State git.latest does not pull latest tags

### ISSUE THREAD

**Issue by jagguli** [saltstack/salt#42329]
### Description of Issue/Question
Given that the local repo and remote repo are both pointing to HEAD, on same branch 
When I tag HEAD on remote repo branch
And execute git.latest with fetch_tags=True
Then local repo does not have the new tag.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report

```/opt/salt:» salt --versions-report                                                                                                                                                                                     [130] [master*] 12:44:35
Salt Version:
           Salt: 2016.11.6

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 3.2.3
       dateutil: 2.6.1
      docker-py: 1.10.6
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: 0.3.6
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.6

System Versions:
           dist: debian 8.8
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.8
```

**Comment by gtmanfred** [comment]
Thanks for the PR!

Daniel

**Comment by gtmanfred** [comment]
If you want this fixed in older versions, you will need to change the PR to be against the 2016.11 branch so that it can be merged forward to the newer versions.

https://docs.saltstack.com/en/latest/topics/development/contributing.html#which-salt-branch

Thanks,
Daniel

**Comment by jagguli** [comment]
@gtmanfred I'm fine with it going in develop, and waiting for it in next releasem I'm using a patched version for my needs.

**Comment by gtmanfred** [comment]
sounds good, just want to make sure that i mention that in case people come back and ask why this isn't in 2016.11

Thanks!
Daniel

**Comment by jagguli** [comment]
@gtmanfred I've changed the target to 2016.11, or should it be 2017.7 ?

**Comment by rallytime** [comment]
@jagguli 2016.11 is great. It will be merged-forward to 2017.7 and develop. Thank you for your contribution.
